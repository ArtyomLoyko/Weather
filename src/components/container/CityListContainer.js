import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toggleDropdown, changeInputValue } from './../../actions';

import CityListItem from '../presentational/CityListItem';

const Input = styled.input`
    font-size: 1rem;
    outline: none;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.3);
    margin-left: 5px;
    text-indent: 2px;
    user-select: none;

    &:hover {
        background-color: rgba(240, 240, 240, 0.8);
    }

    &::-webkit-input-placeholder {
        font-size: 0.85rem;
        font-style: italic;
        padding-left: 3px;
    }

    &:focus::-webkit-input-placeholder {
        color: transparent;
    }
`;

const Container = styled.ul`
    position: absolute;
    top: 20px;
    left: 0;
    display: flex;
    flex-direction: column;
    width: 100%;
    list-style-type: none;
    padding-left: 0;
    border: 1px solid rgba(0, 0, 0, 0.15);
    background: white;
    border-radius: 5px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
`;

export class CityListContainer extends Component {
    handleItemClick = (e, value) => {
        this.props.toggleDropdown(false);
        // this.props.changeInputValue(value);
        // console.log(this.input)
        this.input.value = value;
    };

    render() {
        const {
            citiesData,
            handleChangeInput,
            inputValue,
            dropdownIsOpen,
        } = this.props;
        return (
            <Fragment>
                <Input
                    /*value={inputValue} */ onChange={handleChangeInput}
                    innerRef={el => (this.input = el)}
                />
                {citiesData.length > 0 &&
                    dropdownIsOpen && (
                        <Container>
                            {citiesData.map(city => (
                                <CityListItem
                                    data={city.value}
                                    onClick={e =>
                                        this.handleItemClick(e, city.value)
                                    }
                                />
                            ))
                            /*) : (
          <CityListItem
            data="No such city"
            //handleSubmit={event => this.handleSubmit(event)}
          />*/
                            }
                        </Container>
                    )}
            </Fragment>
        );
    }
}

CityListContainer.propTypes = {
    citiesData: PropTypes.array,
};

export default connect(
    state => ({
        citiesData: state.map.citiesData,
        dropdownIsOpen: state.map.dropdownIsOpen,
        // inputValue: state.map.inputValue
    }),
    { toggleDropdown, changeInputValue }
)(CityListContainer);
