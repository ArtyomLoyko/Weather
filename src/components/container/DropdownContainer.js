import React, { Component } from 'react';
import onClickOutside from 'react-onclickoutside';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toggleDropdown, changeInputValue } from './../../actions';

import CityListItem from '../presentational/CityListItem';

const Container = styled.ul`
    position: absolute;
    top: 20px;
    left: 0;
    display: flex;
    flex-direction: column;
    width: 100%;
    list-style-type: none;
    padding-left: 0;
    border: 1px solid rgba(0, 0, 0, 0.15);
    background: white;
    border-radius: 5px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
`;

export class CityListContainer extends Component {
    handleItemClick = (e, value) => {
        this.props.toggleDropdown(false);
        // this.props.changeInputValue(value);
        // console.log(this.input)
        this.input.value = value;
    };

    render() {
        const {
            citiesData,
            handleChangeInput,
            inputValue,
            dropdownIsOpen,
        } = this.props;
        return (
            citiesData.length > 0 &&
            dropdownIsOpen && (
                <Container>
                    {citiesData.map(city => (
                        <CityListItem
                            data={city.value}
                            onClick={e => this.handleItemClick(e, city.value)}
                        />
                    ))
                    /*) : (
          <CityListItem
            data="No such city"
            //handleSubmit={event => this.handleSubmit(event)}
          />*/
                    }
                </Container>
            )
        );
    }
}

CityListContainer.propTypes = {
    citiesData: PropTypes.array,
};

export default connect(
    state => ({
        citiesData: state.map.citiesData,
        dropdownIsOpen: state.map.dropdownIsOpen,
        // inputValue: state.map.inputValue
    }),
    { toggleDropdown, changeInputValue }
)(onClickOutside(CityListContainer));
