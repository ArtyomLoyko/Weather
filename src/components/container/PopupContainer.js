import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Popup } from 'react-map-gl';

import { closePopup } from '../../actions';

import Weather from '../presentational/Weather';

class PopupContainer extends Component {
    render() {
        const { popup, city, weather, isError, closePopup } = this.props;

        return !popup.isOpen ? null : (
            <Popup
                latitude={popup.coord[1]}
                longitude={popup.coord[0]}
                closeButton={true}
                onClose={closePopup}
                tipSize={25}
                anchor={'top-left'}
                offsetLeft={15}
                offsetTop={15}
            >
                {isError ? (
                    <p>Sorry, there is no weather forecast for your city...</p>
                ) : (
                    <Weather city={city} weather={weather} />
                )}
            </Popup>
        );
    }
}

PopupContainer.propTypes = {
    city: PropTypes.string,
    weather: PropTypes.array,
    popup: PropTypes.object.isRequired,
    closePopup: PropTypes.func.isRequired,
};

export default connect(
    state => ({
        popup: state.popup,
        city: state.weather.city,
        weather: state.weather.data,
        isError: state.weather.isError,
    }),
    { closePopup }
)(PopupContainer);
