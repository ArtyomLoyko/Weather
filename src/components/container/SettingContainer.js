import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchCoord, toggleDropdown } from '../../actions';

import Setting from '../presentational/Setting';

export class SettingContainer extends Component {
    handleChangeInput = e => {
        this.props.fetchCoord(e.target.value);
        this.props.toggleDropdown(true);
    };

    handleSubmit = e => {
        e.preventDefault();
        this.props.getForecast();
    };

    render() {
        return (
            <Setting
                handleChangeInput={this.handleChangeInput}
                handleSubmit={e => this.handleSubmit(e)}
            />
        );
    }
}

SettingContainer.propTypes = {
    isError: PropTypes.bool,
    fetchCoord: PropTypes.func.isRequired,
};

export default connect(
    state => ({
        isError: state.map.isError,
        inputValue: state.map.inputValue,
    }),
    { fetchCoord, toggleDropdown }
)(SettingContainer);
