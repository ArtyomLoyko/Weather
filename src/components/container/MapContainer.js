import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MapGL, { NavigationControl, FlyToInterpolator } from 'react-map-gl';
import { onChangeViewport } from 'redux-map-gl';

import { MAPBOX_TOKEN } from '../../constants';

import PopupContainer from './PopupContainer';

class MapContainer extends Component {
    render() {
        const { viewport, onChangeViewport } = this.props;
        return (
            <MapGL
                {...viewport}
                transitionInterpolator={new FlyToInterpolator()}
                mapStyle="mapbox://styles/mapbox/streets-v9"
                onViewportChange={onChangeViewport}
                mapboxApiAccessToken={MAPBOX_TOKEN}
            >
                <div style={{ position: 'fixed', right: 15, top: 15 }}>
                    <NavigationControl onViewportChange={onChangeViewport} />
                </div>
                <PopupContainer
                    latitude={viewport.latitude}
                    longitude={viewport.longitude}
                />
            </MapGL>
        );
    }
}

MapContainer.propTypes = {
    viewport: PropTypes.object.isRequired,
    onChangeViewport: PropTypes.func.isRequired,
};

export default connect(
    state => ({
        viewport: state.map.viewport.toJS(),
    }),
    { onChangeViewport }
)(MapContainer);
