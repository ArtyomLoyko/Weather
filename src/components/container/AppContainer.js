import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';

import { toggleWarning } from './../../actions';

import MapContainer from './MapContainer';
import SettingContainer from './SettingContainer';
import ErrorCityTooltip from './../presentational/ErrorCityTooltip';

import './../../styles/normalize.css';

export class AppContainer extends Component {
  render() {
    const { warningIsOpen, toggleWarning } = this.props;
    return (
      <Fragment>
        <SettingContainer />
        <MapContainer />
        {warningIsOpen && <ErrorCityTooltip toggleWarning={toggleWarning} />}
      </Fragment>
    );
  }
}

export default connect(
  state => ({
    warningIsOpen: state.map.warningIsOpen,
  }),
  { toggleWarning }
)(AppContainer);
