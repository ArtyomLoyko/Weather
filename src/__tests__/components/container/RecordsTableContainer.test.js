import React from 'react';
import { shallow } from 'enzyme';
import { RecordsTableContainer } from './../../../components/container/RecordsTableContainer';

describe('RecordsTableContainer', () => {
    const props = {
        results: [
            {
                _id: '1',
                username: 'alex',
                email: 'alex@gmail.com',
                score: 25,
            },
            {
                _id: '2',
                username: 'den',
                email: 'den@gmail.com',
                score: 35,
            },
        ],
        isError: false,
    };

    describe('RecordsTableContainer render results', () => {
        const recordsTableContainer = shallow(
            <RecordsTableContainer {...props} />
        );

        it('renders properly', () => {
            expect(recordsTableContainer).toMatchSnapshot();
        });

        it('render results', () => {
            expect(recordsTableContainer.find('RecordsTableRow')).toHaveLength(
                2
            );
        });
    });

    describe('RecordsTableContainer render without results', () => {
        const nextProps = {
            ...props,
            results: null,
            isError: true,
        };

        const recordsTableContainer = shallow(
            <RecordsTableContainer {...nextProps} />
        );

        it('renders properly', () => {
            expect(recordsTableContainer).toMatchSnapshot();
        });

        it('render no results', () => {
            expect(recordsTableContainer.find('RecordsTableRow')).toHaveLength(
                0
            );
        });
    });
});
