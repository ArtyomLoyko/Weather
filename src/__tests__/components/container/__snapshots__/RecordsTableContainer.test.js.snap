// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordsTableContainer RecordsTableContainer render results renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecordsTableContainer
    isError={false}
    results={
        Array [
            Object {
              "_id": "1",
              "email": "alex@gmail.com",
              "score": 25,
              "username": "alex",
            },
            Object {
              "_id": "2",
              "email": "den@gmail.com",
              "score": 35,
              "username": "den",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
                    <th>
                              User Name
                    </th>
                    <th>
                              Email
                    </th>
                    <th>
                              Score
                    </th>
          </tr>
</thead>,
        <tbody>
          <RecordsTableRow
                    email="alex@gmail.com"
                    score={25}
                    username="alex"
          />
          <RecordsTableRow
                    email="den@gmail.com"
                    score={35}
                    username="den"
          />
</tbody>,
      ],
      "className": "table-records",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
                        User Name
            </th>
            <th>
                        Email
            </th>
            <th>
                        Score
            </th>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                User Name
</th>,
              <th>
                Email
</th>,
              <th>
                Score
</th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "User Name",
              },
              "ref": null,
              "rendered": "User Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Score",
              },
              "ref": null,
              "rendered": "Score",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <RecordsTableRow
              email="alex@gmail.com"
              score={25}
              username="alex"
/>,
            <RecordsTableRow
              email="den@gmail.com"
              score={35}
              username="den"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "email": "alex@gmail.com",
              "score": 25,
              "username": "alex",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "email": "den@gmail.com",
              "score": 35,
              "username": "den",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
                        <th>
                                    User Name
                        </th>
                        <th>
                                    Email
                        </th>
                        <th>
                                    Score
                        </th>
            </tr>
</thead>,
          <tbody>
            <RecordsTableRow
                        email="alex@gmail.com"
                        score={25}
                        username="alex"
            />
            <RecordsTableRow
                        email="den@gmail.com"
                        score={35}
                        username="den"
            />
</tbody>,
        ],
        "className": "table-records",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                            User Name
              </th>
              <th>
                            Email
              </th>
              <th>
                            Score
              </th>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  User Name
</th>,
                <th>
                  Email
</th>,
                <th>
                  Score
</th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "User Name",
                },
                "ref": null,
                "rendered": "User Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Score",
                },
                "ref": null,
                "rendered": "Score",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RecordsTableRow
                email="alex@gmail.com"
                score={25}
                username="alex"
/>,
              <RecordsTableRow
                email="den@gmail.com"
                score={35}
                username="den"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "email": "alex@gmail.com",
                "score": 25,
                "username": "alex",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "email": "den@gmail.com",
                "score": 35,
                "username": "den",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`RecordsTableContainer RecordsTableContainer render without results renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecordsTableContainer
    isError={true}
    results={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
                    <th>
                              User Name
                    </th>
                    <th>
                              Email
                    </th>
                    <th>
                              Score
                    </th>
          </tr>
</thead>,
        <tbody>
          <tr>
                    <td
                              colSpan="3"
                    >
                              Error while loading highscores, the server doesn't work...
                    </td>
          </tr>
</tbody>,
      ],
      "className": "table-records",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
                        User Name
            </th>
            <th>
                        Email
            </th>
            <th>
                        Score
            </th>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                User Name
</th>,
              <th>
                Email
</th>,
              <th>
                Score
</th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "User Name",
              },
              "ref": null,
              "rendered": "User Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Score",
              },
              "ref": null,
              "rendered": "Score",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <td
                        colSpan="3"
            >
                        Error while loading highscores, the server doesn't work...
            </td>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <td
              colSpan="3"
>
              Error while loading highscores, the server doesn't work...
</td>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Error while loading highscores, the server doesn't work...",
              "colSpan": "3",
            },
            "ref": null,
            "rendered": "Error while loading highscores, the server doesn't work...",
            "type": "td",
          },
          "type": "tr",
        },
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
                        <th>
                                    User Name
                        </th>
                        <th>
                                    Email
                        </th>
                        <th>
                                    Score
                        </th>
            </tr>
</thead>,
          <tbody>
            <tr>
                        <td
                                    colSpan="3"
                        >
                                    Error while loading highscores, the server doesn't work...
                        </td>
            </tr>
</tbody>,
        ],
        "className": "table-records",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                            User Name
              </th>
              <th>
                            Email
              </th>
              <th>
                            Score
              </th>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  User Name
</th>,
                <th>
                  Email
</th>,
                <th>
                  Score
</th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "User Name",
                },
                "ref": null,
                "rendered": "User Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Score",
                },
                "ref": null,
                "rendered": "Score",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <td
                            colSpan="3"
              >
                            Error while loading highscores, the server doesn't work...
              </td>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <td
                colSpan="3"
>
                Error while loading highscores, the server doesn't work...
</td>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error while loading highscores, the server doesn't work...",
                "colSpan": "3",
              },
              "ref": null,
              "rendered": "Error while loading highscores, the server doesn't work...",
              "type": "td",
            },
            "type": "tr",
          },
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
